File: app/src/main/java/deltazero/amarok/AppHider/NoneAppHider.java
package deltazero.amarok.AppHider;

import android.content.Context;
import android.util.Log;

import java.util.Set;

public class NoneAppHider extends AppHiderBase{
    
    // Active when no AppHider is available.

    public NoneAppHider(Context context) {
        super(context);
    }

    @Override
    public void hide(Set<String> pkgNames) {
        Log.w("AppHider", "Skip app hiding: hider disabled");
    }

    @Override
    public void unhide(Set<String> pkgNames) {
        Log.w("AppHider", "Skip app unhiding: hider disabled");
    }

    @Override
    public CheckAvailabilityResult checkAvailability() {
        return new CheckAvailabilityResult(CheckAvailabilityResult.Result.AVAILABLE);
    }

    @Override
    public void active(OnActivateCallbackListener onActivateCallbackListener) {
        onActivateCallbackListener.onActivateCallback(getClass(), true, 0);
    }

    @Override
    public String getName() {
        return "None";
    }
}

File: app/src/main/java/deltazero/amarok/apphider/NoneAppHider.kt
package deltazero.amarok.apphider

import android.util.Log
import deltazero.amarok.core.ActivationResult
import deltazero.amarok.core.Hider

class NoneAppHider : AppHider {
  override val mode = AppHiderMode.NONE
  override val name = "Disabled"

  override suspend fun activate() = ActivationResult(success = true, msgResId = 0)

  override suspend fun process(pkgNames: Set<String>, action: Hider.Action) {
    Log.w(
      "AppHider",
      "Skip app ${if (action == Hider.Action.HIDE) "hiding" else "unhiding"}: hider disabled",
    )
  }
}
